src/scripts/alleles/G_domain.py

```diff
@@ -0,0 +1,128 @@
+            exon_3 = exons[2].extract(entry)
+            G_domain_nuc = G_domain_nuc + exon_3
+        elif type == 4 and len(exons) == 2:
+            exon_1_len = exons[0].location.end - exons[0].location.start
+            last_full_codon_start = list(range(codon_start - 1, exon_1_len -1, 3))[-1]
+            splice_loc = last_full_codon_start + 2 + 1
+            splice = exons[0].extract(entry)[splice_loc: ]
+            exon_2 = exons[1].extract(entry)
+            G_domain_nuc = splice + exon_2
+        elif type == 2 and len(exons) == 2 and has_exon_1:
+            exon_1_len = exons[0].location.end - exons[0].location.start
+            if exon_1_len == 1:
+                splice_loc = 0
+            else:
+                last_full_codon_start = list(range(codon_start - 1, exon_1_len -1, 3))[-1]
+                splice_loc = last_full_codon_start + 2 + 1
+            splice = exons[0].extract(entry)[splice_loc: ]
+            exon_2 = exons[1].extract(entry)
+            G_domain_nuc = splice + exon_2
+        elif type == 2 and len(exons) == 1 and not has_exon_1:
+            G_domain = cds.qualifiers['translation'][0]
+        elif type == 1 and len(exons) == 3 and has_exon_1:
+            exon_1_len = exons[0].location.end - exons[0].location.start
+            if exon_1_len == 1:
+                splice_loc = 0
+            else:
+                last_full_codon_start = list(range(codon_start - 1, exon_1_len -1, 3))[-1]
+                splice_loc = last_full_codon_start + 2 + 1
+            splice = exons[0].extract(entry)[splice_loc: ]
+            exon_2 = exons[1].extract(entry)
+            G_domain_nuc = splice + exon_2
+            exon_3 = exons[2].extract(entry)
+            G_domain_nuc = G_domain_nuc + exon_3
+        elif type == 1 and len(exons) == 2 and not has_exon_1:
+            G_domain = cds.qualifiers['translation'][0]
+        else:
+            #import pdb; pdb.set_trace()
+            excluded_sequence.append(entry.name)
+            continue
+        if not G_domain:
+            G_domain = G_domain_nuc.translate()
+        if 'seq' in dir(G_domain) and str(G_domain.seq) not in cds.qualifiers['translation'][0]:
+            import pdb; pdb.set_trace()
+            print(entry.name, G_domain.seq, cds.qualifiers['translation'][0], "not matched" )
+        if 'seq' in dir(G_domain):
+            G_domain =  str(G_domain.seq)
+        else:
+            G_domain = G_domain
+        G_domains.append({"Label": data[entry.name], "minimal G domain sequence" : G_domain})
+    except BiopythonWarning:
+        print("BiopythonWarning")
+    except AttributeError:
+        if str(entry.seq) == 'X':
+            excluded_sequence.append(entry)
+
+print(excluded_sequence)
+
+import csv
+with open("ontology/G-domain-sequence.tsv", "w") as fh:
+    writer = csv.DictWriter(fh, fieldnames = G_domains[0].keys(), delimiter = "\t")
+    writer.writeheader()
+    fh.write("LABEL\tA minimal G domain sequence")
```

Comment:
This needs a `\n` at the end of this line, otherwise your first row ends up on the same line as the ROBOT template strings.

Reply:
This should also match the label in `ontology/properties.tsv`, which is "minimal HLA G domain sequence".